src/partners/transak.ts

```diff
@@ -0,0 +1,101 @@
+import { asArray, asNumber, asObject, asOptional, asString, asBoolean, asUnknown, asUndefined, asNull } from 'cleaners'
+import fetch from 'node-fetch'
+import { PartnerPlugin, PluginParams, PluginResult, StandardTx } from '../types'
+
+const pageLimit = 100
+
+const asTransakOrder = asObject({
+    status: asString,
+    id: asString,
+    fromWalletAddress: asOptional,
+    fiatCurrency: asString,
+    fiatAmount: asNumber,
+    walletAddress: asString,
+    cryptocurrency: asString,
+    cryptoAmount: asNumber,
+    completedAt: asOptional(asString)
+})
+
+const asTransakResult = asObject({
+    response: asArray(asUnknown)
+})
+
+
+export async function queryTransak(
+    pluginParams: PluginParams
+): Promise<PluginResult> {
+    const ssFormatTxs: StandardTx[] = []
+    let apiKey: string
+
+    let { offset = 0 } = pluginParams.settings
+    if (typeof pluginParams.apiKeys.transak_api_secret === 'string') {
+        apiKey = pluginParams.apiKeys.transak_api_secret
+    } else {
+        return {
+            settings: { offset: offset },
+            transactions: []
+        }
+    }
+
+    let resultJSON
+    let done = false
+
+    while (!done) {
+        let url = `https://api.transak.com/api/v1/partners/orders/?partnerAPISecret=${apiKey}&limit=${pageLimit}&skip=${offset}`
+        let jsonObj: ReturnType<typeof asTransakResult>
+        try {
+            const result = await fetch(url)
+            resultJSON = await result.json()
+            jsonObj = asTransakResult(resultJSON)
+        } catch (e) {
+            console.log(e)
+            break
+        }
+        const txs = jsonObj.response
+        for (const rawtx of txs) {
+            let tx
+            try {
+                tx = asTransakOrder(rawtx)
+            } catch (e) {
+                console.log(e)
+                continue
+            }
+            if (tx.status === 'COMPLETED') {
```

Comment:
inbetween line 63-64 put const date = new Date(tx.completedAt), and then replace date with those two instances on line 73 and 74